### PR TITLE
Foreign key enforcement added on database level

### PR DESCRIPTION
**Pull request by asif-fndr**
Before this fix: there was no database level foreign key enforcement happening even if we create a reference using `orm:"rel(fk);null"`
After this fix: the `orm:"rel(fk);null"` adds the foreign key enforcement on database level.

### CODE REVIEW

**Comment by asif-fndr** [comment]
@astaxie pls review

**Comment by astaxie** [comment]
Could you add test case for this change

**Comment by astaxie** [comment]
@asif-fndr Could you add the test case for it?

**Comment by astaxie** [comment]
@asif-fndr Could you add the test for it? I will release a new version. If you can add the test case before that release would be great.

**Comment by asif-fndr** [comment]
@astaxie added unit test . :56f97866c0c996dc6697f00b73cd11c60aebf273
thought i noticed that there are issues in general in orm_test.go; not clear how we ensure that all models are registered before running UTs and thats why it fails to create tables.

File: orm/utils.go [hunk 1]
```diff
@@ -257,6 +257,13 @@ func SetNameStrategy(s string) {
 	nameStrategy = s
 }
 
+// converst *pkg1.pkg2.StructName to StructName
```
Comment (gogitops) [inline orm/utils.go:260]: warning: "converst" is a misspelling of "converts" (misspell)

**Comment by flycash** [comment]
It's inactive for a long time. I closed it. If necessary, we could move it to develop-2.0.

**Pull request closed without merging**